### PR TITLE
Fix/crop transforms

### LINKED ISSUE #1735: RandomCrop, CenterCrop and RandomResizeCrop doesn't crop bbox or keypoints when cropping_mode="slice"
### Describe the bug

RandomCrop, CenterCrop and RandomResizeCrop doesn't crop bbox or keypoint inputs when cropping_mode is "slice".  Mask inputs works fines.

Note that using cropping_mode="resample" works fine. 

### Reproduction steps

```bash
Download panda image: `wget https://tinypng.com/images/social/website.jpg -O panda.jpg`


import torch
import kornia.augmentation as K
import torchvision
from kornia.utils import draw_rectangle, tensor_to_image

def plot(image, bbox, mask=None):
    if mask is not None:
        image *= mask.any(dim=1) # Collapse all masks from different boxes
    img_draw = draw_rectangle(image.clone().detach(), bbox, color=torch.as_tensor([255., 0, 0]))
    plt.imshow(tensor_to_image(img_draw)); plt.axis('off');

# Inputs
image = torchvision.io.read_image('panda.jpg').unsqueeze(0).float() / 255.
boxes = torch.tensor([[[355,10, 660,250], [10,125, 200,400]]])
mask = Boxes.from_tensor(boxes, mode="xyxy").to_mask(*image.shape[-2:])

plot(image, boxes)

aug = K.AugmentationSequential(
   K.RandomCrop((500,500), p=1, cropping_mode="slice"), # <- works with cropping_mode="resampling"
   data_keys=["input", "bbox_xyxy", "mask"],  
)

tfm_img, tfm_bbox, tfm_mask = aug(image, boxes, mask)
```

### Expected behavior

Do crop for bbox and keypoints like any other transform does. 

### Environment

```shell
Kornia 0.6.5
Pytorch 1.9.1+cpu
```

### Additional context

_No response_

*Edit: add RandomResizeCrop and remove 3D version of these transforms since they aren't affected*

### PR DESCRIPTION
**Pull request by hal-314**
#### Changes

Fixes #1735

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by shijianjian — APPROVED** [review]
This can be a temp fix since the slice mode shall not compute any transformation matrices at all. 

Probably worth to revisit later.

**Review by edgarriba — APPROVED** [review]
